contracts/Core/StakeManager.sol

```diff
@@ -173,44 +176,65 @@ contract StakeManager is Initializable, StakeStorage, StateManager, Pause, Stake
     /// @param stakerId The Id of staker associated with sRZR which user want to unstake
     /// @param sAmount The Amount in sRZR
     function unstake(uint32 stakerId, uint256 sAmount) external initialized whenNotPaused {
+        require(sAmount > 0, "Non-Positive Amount");
+        require(stakerId != 0, "staker.id = 0");
+        require(stakers[stakerId].stake > 0, "Nonpositive stake");
+        require(locks[msg.sender][stakers[stakerId].tokenAddress][LockType.Unstake].amount == 0, "Existing Unstake Lock");
+
+        uint32 epoch = _getEpoch(epochLength);
+        Structs.Staker storage staker = stakers[stakerId];
+        IStakedToken sToken = IStakedToken(staker.tokenAddress);
+
+        require(sToken.balanceOf(msg.sender) >= sAmount, "Invalid Amount");
+
+        locks[msg.sender][staker.tokenAddress][LockType.Unstake] = Structs.Lock(sAmount, 0, epoch + unstakeLockPeriod);
+        emit Unstaked(msg.sender, epoch, stakerId, sAmount, staker.stake, block.timestamp);
+
+        require(sToken.transferFrom(msg.sender, address(this), sAmount), "sToken transfer failed");
+    }
+
+    function initiateWithdraw(uint32 stakerId) external initialized whenNotPaused {
         State currentState = _getState(epochLength);
         require(currentState != State.Propose, "Unstake: NA Propose");
         require(currentState != State.Dispute, "Unstake: NA Dispute");
+
         uint32 epoch = _getEpoch(epochLength);
+        require(stakerId != 0, "staker doesnt exist");
+        require(locks[msg.sender][stakers[stakerId].tokenAddress][LockType.Unstake].unlockAfter != 0, "Did not unstake");
+        require(locks[msg.sender][stakers[stakerId].tokenAddress][LockType.Unstake].unlockAfter <= epoch, "Withdraw epoch not reached");
+        require(
+            locks[msg.sender][stakers[stakerId].tokenAddress][LockType.Unstake].unlockAfter + withdrawInitiationPeriod >= epoch,
+            "Initiation Period Passed"
+        ); // Can Use ExtendLock
 
         Structs.Staker storage staker = stakers[stakerId];
-        require(staker.id != 0, "staker.id = 0");
-        require(staker.stake > 0, "Nonpositive stake");
-        require(locks[msg.sender][staker.tokenAddress].amount == 0, "Existing Lock");
-        require(sAmount > 0, "Non-Positive Amount");
+        IStakedToken sToken = IStakedToken(staker.tokenAddress);
+        Structs.Lock storage lock = locks[msg.sender][staker.tokenAddress][LockType.Unstake];
```

Comment:
can we move this lines variable lines up to this requires 
and in req use those variables 
that would be more efficient and clean imo
```
 require(locks[msg.sender][stakers[stakerId].tokenAddress][LockType.Unstake].unlockAfter != 0, "Did not unstake");

  ```